Patch by Senquack

--- a/src/audio/alsa/SDL_alsa_audio.c
+++ b/src/audio/alsa/SDL_alsa_audio.c
@@ -478,6 +478,10 @@
 		SDL_SetError("Couldn't open audio device: %s", SDL_NAME(snd_strerror)(status));
 		return(-1);
 	}
+	
+	/* Switch to blocking mode for playback */
+	/* Note: this must happen before hw/sw params are set. */
+	SDL_NAME(snd_pcm_nonblock)(pcm_handle, 0);
 
 	/* Figure out what the hardware is capable of */
 	snd_pcm_hw_params_alloca(&hwparams);
@@ -611,9 +615,6 @@
 	}
 	SDL_memset(mixbuf, spec->silence, spec->size);
 
-	/* Switch to blocking mode for playback */
-	SDL_NAME(snd_pcm_nonblock)(pcm_handle, 0);
-
 	/* We're ready to rock and roll. :-) */
 	return(0);
 }
